app/scripts/components/table/list-view.js

```diff
@@ -21,63 +26,61 @@ class List extends React.Component {
     this.updateSelection = this.updateSelection.bind(this);
     this.onBulkActionSuccess = this.onBulkActionSuccess.bind(this);
     this.onBulkActionError = this.onBulkActionError.bind(this);
-    this.config = this.config.bind(this);
+    this.getQueryConfig = this.getQueryConfig.bind(this);
     this.renderSelectAll = this.renderSelectAll.bind(this);
+
+    const initialPage = 1;
+    const initialSortIdx = props.sortIdx || 0;
+    const initialOrder = 'desc';
+
     this.state = {
-      page: 1,
-      sortIdx: props.sortIdx || 0,
-      order: 'desc',
+      page: initialPage,
+      sortIdx: initialSortIdx,
+      order: initialOrder,
       selected: [],
       prefix: null,
-      queryConfig: {},
+      queryConfig: {
+        page: initialPage,
+        order: initialOrder,
+        sort_by: this.getSortProp(initialSortIdx),
+        params: {},
+        ...(props.query || {})
+      },
       params: {},
       completedBulkActions: 0,
       bulkActionError: null
     };
   }
 
-  componentDidMount () {
-    this.setState({ queryConfig: this.config() }); // eslint-disable-line react/no-did-mount-set-state
-  }
-
-  componentDidUpdate (prevProps) {
+  componentDidUpdate (prevProps, prevState) {
     const { query, list, sortIdx } = this.props;
-    if (JSON.stringify(query) !== JSON.stringify(prevProps.query)) {
-      this.setState({ queryConfig: this.config({}, query) }); // eslint-disable-line react/no-did-update-set-state
-    }
-
-    // remove null and undefined values
-    const { params } = list;
-    const validParams = {};
-    for (let key in params) {
-      let value = params[key];
-      if (!undef(value) && value !== null) {
-        validParams[key] = value;
-      }
-    }
-
-    if (JSON.stringify(validParams) !== JSON.stringify(this.state.params)) {
-      this.setState({ params: validParams }, () => this.setState({ // eslint-disable-line react/no-did-update-set-state
```

Comment:
Hmm, why was it waiting to update the queryConfig state until after the params state was updated? Is it bad that it doesn't work that way now?

Reply:
Actually, it does still work that way in essence. However, with the refactoring, only 1 call to `setState` is necessary instead of two. The incremental state object is combined by the new lines of code so that only 1 call is necessary. See the new lines 65-68.